src/discussion/views.py

```diff
@@ -226,6 +227,30 @@ def upvote(self, *args, **kwargs):
     def downvote(self, *args, **kwargs):
         return super().downvote(*args, **kwargs)
 
+    @action(
+        detail=True,
+        methods=["post"],
+        permission_classes=[IsOriginalQuestionPoster],
+    )
+    def mark_as_accepted_answer(self, *args, **kwargs):
+        try:
+            document_id = get_document_id_from_path(self.request)
+            target_post_question = ResearchhubPost.objects.get(id=document_id)
+
+            # logical ordering - DO NOT CHANGE THE ORDER OF OPERATIONS
+            target_thread = self.get_object()
+            target_thread.is_accepted_answer = True
+            target_thread.save()
+
+            current_accepted_answer = target_post_question.get_accepted_answer()
```

Comment:
Can we change to `prev_accepted_answer`? `current` is confusing.

I think since we save the other already, it is okay to consider it as `prev`